jax_dna/optimization/objective.py

```diff
@@ -113,7 +113,7 @@ def update(
             for exposed, output in filter(
                 lambda e: e[0] in self._needed_observables, zip(sim_exposes, sim_output, strict=True)
             ):
-                self._obtained_observables.append((exposed, jdna_tree.load_pytree(output)))
```

Comment:
My original thought with writing the trajectories/observables to disk is that in the case distributed simulators/objectives, large trajectories and observables need to be passed back to the coordinating node and to the other nodes that need trajectory/observable. If we write them to disk then we aren't asking the coordinating node to hold all the data and to send it back to the other nodes, hopefully in cases like this we're also writing to high performance shared memory.

However, if you think if this might be a bit of overengineering or just not a best practice, then I am happy to switch.